sstables/sstable_directory.cc

```diff
@@ -642,16 +634,19 @@ future<> sstable_directory::filesystem_components_lister::garbage_collect(storag
 future<> sstable_directory::filesystem_components_lister::cleanup_column_family_temp_sst_dirs() {
     std::vector<future<>> futures;
 
-    co_await lister::scan_dir(_directory, lister::dir_entry_types::of<directory_entry_type::directory>(), [&] (fs::path sstdir, directory_entry de) {
-        // push futures that remove files/directories into an array of futures,
-        // so that the supplied callback will not block scan_dir() from
-        // reading the next entry in the directory.
-        fs::path dirpath = sstdir / de.name;
-        if (dirpath.extension().string() == tempdir_extension) {
-            sstlog.info("Found temporary sstable directory: {}, removing", dirpath);
-            futures.push_back(io_check([dirpath = std::move(dirpath)] () { return lister::rmdir(dirpath); }));
+    directory_lister lister(_directory, lister::dir_entry_types::of<directory_entry_type::directory>());
+    co_await with_closeable(std::move(lister), [this, &f = futures] (directory_lister& lister) -> future<> {
+        auto& futures = f;
```

Comment:
How about collecting `futures` locally in the coroutine lambda and co_return it
instead of using a variable in the caller's scope (with capture by reference)?

Reply:
You mean the lambda inside with_closeable()? In that case the object will be left unclosed for longer than it's really needed

Reply:
Ah, now I see your point (after a sleep :yawning_face: ). Nice suggestion, will try